packages/slice-machine/lib/models/common/widgets/Image/components.tsx

```diff
@@ -133,54 +120,53 @@ export const ConstraintForm: React.FC<{
 };
 
 const TexField: React.FC<{
-  prefix: string;
   name: string;
   label: string;
   placeholder: string;
-  onChangeSetField: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  isConstraint: boolean;
+  meta: FieldMetaProps<string>;
   sx?: ThemeUIStyleObject;
-}> = ({ prefix, name, label, placeholder, onChangeSetField, ...props }) => {
-  const fieldName = `${prefix}.${name}`;
+}> = ({ name, label, placeholder, meta, isConstraint, ...props }) => {
+  const [field] = useField<string>(name);
 
-  const [field, meta] = useField<string>(fieldName);
   return (
     <FormFieldInput
       {...props}
-      fieldName={fieldName}
+      fieldName={name}
       meta={meta}
       formField={{ label, placeholder }}
       field={
-        prefix === "config.constraint"
+        isConstraint
           ? { value: "main", readOnly: true }
-          : { value: field.value, onChange: onChangeSetField }
+          : { value: field.value }
       }
-      variant={prefix === "config.constraint" ? "disabled" : "primary"}
+      variant={isConstraint ? "disabled" : "primary"}
     />
   );
 };
 
 const NumberField: React.FC<{
-  prefix: string;
   name: string;
   label: string;
   placeholder: string;
-  onChangeSetField: (e: React.ChangeEvent<HTMLInputElement>) => void;
   sx?: ThemeUIStyleObject;
-}> = ({ prefix, name, label, placeholder, onChangeSetField, ...props }) => {
-  const fieldName = `${prefix}.${name}`;
-
-  const [field, meta] = useField<number>(fieldName);
-
+}> = ({ name, label, placeholder, ...props }) => {
+  const [field, meta] = useField<string>(name);
   return (
     <FormFieldInput
       {...props}
-      fieldName={fieldName}
-      meta={meta}
+      fieldName={name}
       formField={{ label, placeholder }}
+      meta={meta}
       field={{
+        min: 0,
         type: "number",
         value: field.value || "",
-        onChange: onChangeSetField,
+        onKeyDown: (event) => {
+          if (event.key === "e" || event.key === "-" || event.key === "+") {
```

Comment:
Why would those value be like that ?
feels highly weird to have to prevent keys like that 😓

Reply:
At least, I would prevent everything that is not a number if we really have no other option than using that.

Reply:
thought the E was funny tho 😂